### ISSUE TITLE
Support RegExp `v` flag

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#17223]
### Syntax name

RegExp v flag with set notation + properties of strings

### Syntax proposal URL

https://github.com/tc39/proposal-regexp-v-flag

### Example code

```js
/[\p{Decimal_Number}--[0-9]]/v.test(str);
```

### Implementation Checklist

- [x] Ecma262 update: https://github.com/tc39/ecma262/pull/2418
- [x] ESTree update: N/A
- [x] Acorn update: https://github.com/acornjs/acorn/pull/1218
- [x] `eslint-visitor-keys` update: N/A
- [x] `espree` update: https://github.com/eslint/espree/pull/575
- [x] `regexpp` update: https://github.com/eslint-community/regexpp/pull/82
- [x] `eslint-scope` update: N/A
- [x] `eslint` update:
  - [x] `no-control-regex` https://github.com/eslint/eslint/pull/17405
  - [x] `no-empty-character-class` https://github.com/eslint/eslint/pull/17419
  - [x] `no-invalid-regexp` https://github.com/eslint/eslint/pull/17404
  - [x] `no-misleading-character-class` https://github.com/eslint/eslint/pull/17406
  - [x] `no-regex-spaces` https://github.com/eslint/eslint/pull/17407
  - [x] `no-useless-backreference` https://github.com/eslint/eslint/pull/17408
  - [x] `no-useless-escape` https://github.com/eslint/eslint/pull/17420
  - [x] `prefer-named-capture-group` https://github.com/eslint/eslint/pull/17409
  - [x] `prefer-regex-literals` https://github.com/eslint/eslint/pull/17410
  - [x] `require-unicode-regexp` https://github.com/eslint/eslint/pull/17402

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

prev issue: https://github.com/eslint/eslint/issues/17199

**Comment by mdjermanovic** [comment]
Thanks! Correct, there's no need to update `estree`, `eslint-visitor-keys` and `eslint-scope`. I just added `regexpp` to the list. cc @ota-meshi

**Comment by mdjermanovic** [comment]
I'm working on `no-empty-character-class`.

**Comment by ota-meshi** [comment]
The `no-useless-escape` rule seems to have false positives.

```js
/[\-]/v; // -> Fix to `/[-]/v`, but it SyntaxError
```

I will try to fix it.

**Comment by nzakas** [comment]
It looks like all of the checklist items are complete, so closing.
